app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -25,11 +26,64 @@ def bulk_destroy
         process_response @hosts.destroy_all
       end
 
+      api :PUT, "/hosts/bulk/build", N_("Build")
+      param_group :bulk_host_ids
+      param :reboot, :bool, N_("Reboot after build")
+      param :rebuild_configuration, :bool, N_("Rebuild configuration only")
+      def build
+        if Foreman::Cast.to_bool(params[:rebuild_configuration])
+          rebuild_config
+        else
+          reboot = Foreman::Cast.to_bool(params[:reboot])
+          manager = BulkHostsManager.new(hosts: @hosts)
+          missed_hosts = manager.build(reboot: reboot)
+          if missed_hosts.empty?
+            process_response(true, { :message => _("Built %{host_count} %{hosts}") % { :host_count => @hosts.count, :hosts => 'host'.pluralize(@hosts.count) }})
+          else
+            process_response(false, { :message => _("The following hosts failed the build operation: %s") % missed_hosts.map(&:name).to_sentence})
```

Comment:
Does `to_sentence` actually work with translations?

Reply:
we are not doing `to_sentence` anymore